.github/workflows/build_and_test.yml

```diff
@@ -93,41 +83,59 @@ jobs:
     - name: Check labeler configuration
       run: scripts/check-labeler.pl .github/labeler.yml
 
-    - name: Unit tests
+  schedule:
+    name: Schedule busted tests to run
+    runs-on: ubuntu-22.04
+    needs: build
+
+    env:
+      WORKFLOW_ID: ${{ github.run_id }}
+
+    outputs:
+      runners: ${{ steps.generate-runner-array.outputs.RUNNERS }}
+
+    steps:
+    - name: Checkout source code
+      uses: actions/checkout@v4
+
+    - name: Download runtimes file
+      uses: Kong/gh-storage/download@v1
       env:
-        KONG_TEST_PG_DATABASE: kong
-        KONG_TEST_PG_USER: kong
-      run: |
-          source ${{ env.BUILD_ROOT }}/kong-dev-venv.sh
-          TEST_CMD="bin/busted -v -o htest spec/01-unit"
-          if [[ $KONG_TEST_COVERAGE = true ]]; then
-            TEST_CMD="$TEST_CMD --coverage"
-          fi
-          $TEST_CMD
+        GITHUB_TOKEN: ${{ secrets.PAT }}
+      with:
+        repo-path: Kong/gateway-action-storage/main/.ci/runtimes.json
 
-    - name: Archive coverage stats file
+    - name: Schedule tests
+      uses: Kong/gateway-test-scheduler/schedule@5d27a0bf1acc78fab865eafb41d60e0c7625d261 # v2
+      with:
+        test-suites-file: .ci/test_suites.json
+        test-file-runtime-file: .ci/runtimes.json
+        output-prefix: test-chunk.
+        runner-count: ${{ env.RUNNER_COUNT }}
+        static-mode: ${{ github.run_attempt > 1 }}
+
+    - name: Upload schedule files
       uses: actions/upload-artifact@v4
-      if: ${{ always() && (inputs.coverage == true || github.event_name == 'schedule') }}
+      continue-on-error: true
       with:
-        name: luacov-stats-out-${{ github.job }}-${{ github.run_id }}
-        retention-days: 1
-        path: |
-          luacov.stats.out
+        name: schedule-test-files
+        path: test-chunk.*
+        retention-days: 7
 
-    - name: Get kernel message
-      if: failure()
+    - name: Generate runner array
+      id: generate-runner-array
       run: |
-        sudo dmesg -T
+        echo "RUNNERS=[$(seq -s "," 1 $(( "$RUNNER_COUNT" )))]" >> "$GITHUB_OUTPUT"
 
-  integration-tests-postgres:
-    name: Postgres ${{ matrix.suite }} - ${{ matrix.split }} tests
+  busted-tests:
+    name: Busted test runner ${{ matrix.runner }}
     runs-on: ubuntu-22.04
-    needs: build
+    needs: [build,schedule]
+
     strategy:
       fail-fast: false
       matrix:
-        suite: [integration, plugins]
-        split: [first, second]
+        runner: ${{ fromJSON(needs.schedule.outputs.runners) }}
```

Comment:
Not a blocker: Could we let the scheduler decide how many runners we use?

Reply:
you mean dynamically decide how many runners are needed depending on the amount of tests and maybe other parameters? That could be an interesting feature but for now we are aiming at keeping its logic as simple as possible to avoid making it too smart but unreliable, so the number of runners, which is expected to stay more or less unchanged, is defined statically in this workflow file.